File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        StringBuilder str=new StringBuilder();
        int balance=0;
        for(char ch:s.toCharArray()){
            if(ch=='('){
                if(balance>0) str.append(ch);
                balance++;
            }
            else{
                balance--;
                if(balance>0) str.append(ch);
            }
        }
        return str.toString();
    }
}

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.rs
impl Solution {
    pub fn remove_outer_parentheses(s: String) -> String {
        let mut result=String::from("");
        let mut depth=0;
        for ch in s.chars(){
            if ch=='('{
                if depth>0{
                    result.push(ch);
                }
                depth+=1;
            }
            else {
                depth-=1;
                if depth>0{
                    result.push(ch);
                }
            }
        }
        result
    }
}